infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespace.java

```diff
@@ -141,23 +141,20 @@ void prepare(boolean markManaged, boolean canCreate) throws InfrastructureExcept
 
   /**
    * Deletes the namespace. Deleting a non-existent namespace is not an error as is not an attempt
-   * to delete a namespace that is already being deleted.
+   * to delete a namespace that is already being deleted. If the namespace is not marked as managed,
+   * it is silently not deleted.
```

Comment:
I don't know if silently defaulting to a no-op is the best choice, something should likely be logged.

Reply:
Something is logged :)

Reply:
In that case, "silently" is a bit misleading ;)